contracts/Validate.sol

```diff
@@ -9,25 +9,20 @@ import "./ECRecovery.sol";
  * @dev Checks that the signatures on a transaction are valid
  */
 library Validate {
-    function checkSigs(bytes32 txHash, bytes32 rootHash, uint256 blknum2, bytes sigs)
+    function checkSigs(bytes32 txHash, address exitor, uint256 oindex, bytes sigs)
         internal
         view
         returns (bool)
     {
-        require(sigs.length % 65 == 0 && sigs.length <= 260);
-        bytes memory sig1 = ByteUtils.slice(sigs, 0, 65);
-        bytes memory sig2 = ByteUtils.slice(sigs, 65, 65);
-        bytes memory confSig1 = ByteUtils.slice(sigs, 130, 65);
-        bytes32 confirmationHash = keccak256(txHash, rootHash);
+        require(sigs.length % 65 == 0 && sigs.length <= 130);
 
-        bool check1 = true;
-        bool check2 = true;
-
-        check1 = ECRecovery.recover(txHash, sig1) == ECRecovery.recover(confirmationHash, confSig1);
-        if (blknum2 > 0) {
-            bytes memory confSig2 = ByteUtils.slice(sigs, 195, 65);
-            check2 = ECRecovery.recover(txHash, sig2) == ECRecovery.recover(confirmationHash, confSig2);
+        if (oindex > 0) {
```

Comment:
wouldn't more explicit `if (oindex == 1) else if (oindex == 0) else assert false` consition sequence be more defensive?